src/vaults/PoolManager.sol

```diff
@@ -130,6 +130,8 @@ contract PoolManager is
 
         emit TransferShares(centrifugeId, poolId, scId, msg.sender, receiver, amount);
         sender.sendInitiateTransferShares(poolId, scId, centrifugeId, receiver, amount);
+
+        gateway.endTransactionPayment();
```

Comment:
Post-mortem note. Maybe it makes sense to have a `transactionPay` modifier too, to avoid missing some calling end in some refactor/new feature. I will add to some PR if aggreed

Reply:
Makes sense!